From 8aa77af6f9aad0d8a5b50a0670f85668ad489022 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Fri, 25 May 2012 14:57:38 -0400
Subject: [PATCH 345/364] Don't set boot device on ppc-ieee1275

This started with the problem that powerkvm doesn't have /dev/nvram and so
there is no way to set boot-device.
---
 util/grub-install.in | 18 ++++++++++--------
 1 file changed, 10 insertions(+), 8 deletions(-)

diff --git a/util/grub-install.in b/util/grub-install.in
index 271e447..c9f66c7 100644
--- a/util/grub-install.in
+++ b/util/grub-install.in
@@ -777,14 +777,16 @@ elif [ "${grub_modinfo_target_cpu}-${grub_modinfo_platform}" = "i386-ieee1275" ]
 	    }
 	fi
 
-	"$nvsetenv" boot-device "$boot_device" || {
-	    # TRANSLATORS: The %s will be replaced by an external program name.
-	    gettext_printf "\`%s' failed.\n" "$nvsetenv" 1>&2
-	    gettext "You will have to set \`boot-device' variable manually.  At the IEEE1275 prompt, type:" 1>&2
-	    echo 1>&2
-	    echo "  setenv boot-device $boot_device" 1>&2
-	    exit 1
-	}
+	if [ "${grub_modinfo_target_cpu}-${grub_modinfo_platform}" = "i386-ieee1275" ]; then
+	    "$nvsetenv" boot-device "$boot_device" || {
+	        # TRANSLATORS: The %s will be replaced by an external program name.
+	        gettext_printf "\`%s' failed.\n" "$nvsetenv" 1>&2
+	        gettext "You will have to set \`boot-device' variable manually.  At the IEEE1275 prompt, type:" 1>&2
+	        echo 1>&2
+	        echo "  setenv boot-device $boot_device" 1>&2
+	        exit 1
+	    }
+	fi
     fi
 elif [ x"${grub_modinfo_target_cpu}-${grub_modinfo_platform}" = xmips-arc ]; then
     dvhtool -d "${install_device}" --unix-to-vh "${grubdir}/${grub_modinfo_target_cpu}-$grub_modinfo_platform/core.${imgext}" grub
-- 
1.8.1.4

